[imkwokyong] iP

imkwokyong opened the pull request:
### ForsakenX - A Duke chatbot
> "Legend says this was the origin of ChatGPT." - **Albert Einstein** ([source](https://makeameme.org/meme/dont-believe-everything-4fe13d))

Don't worry about the AI revolution, this chatbot doesn't know jack except simple commands.
However it is a good to-do list:
- It's easy to use 💯 
- ~~Simple~~ No fancy mumbo-jumbos, text-based is king 🥇 
- *F A S T* 👍 

It categorizes your tasks into:
1. `Todo`
2. `Deadline`
3. `Event`

So if you fulfil these criteria:
- [x] Forgetful
- [x] Busy person with many tasks
- [x] No secretary

**THIS CHATBOT IS FOR YOU!**

This bot is programmed using Java. Here's a simple Java code:
```java
public static void main(String[] args) {
    System.out.println("Hello World!");
}
```
[Here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)'s a Java tutorial if you need one!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

imkwokyong committed:
Implement initial skeletal version

imkwokyong committed:
Add Echo

imkwokyong committed:
Add ability to store and display text

imkwokyong committed:
[Level-3] Mark as Done

imkwokyong committed:
Add Deadline.java, Event.java, ToDo.java

imkwokyong committed:
[Level-4] ToDo, Event, Deadline

imkwokyong committed:
[A-TextUiTesting] Automated Text UI Testing

imkwokyong committed:
[Level-5] Handle Errors

imkwokyong committed:
[Level-6] Delete

imkwokyong committed:
Add missing space

imkwokyong committed:
Modify test case to check for exceptions and deletions

imkwokyong committed:
Modify code for conciseness and add extra exceptions

imkwokyong committed:
Modify test case to account for added exceptions

imkwokyong committed:
Fix indentation

imkwokyong committed:
Add constructor overloading for Task.java

imkwokyong committed:
Add string method for easy write access to file

imkwokyong committed:
Update .gitignore

imkwokyong committed:
Add Save function for Level-7

imkwokyong committed:
Merge branch 'branch-Level-7'

imkwokyong committed:
Update .gitignore

imkwokyong committed:
Remove unintended debugging print statement

imkwokyong committed:
Add Dates and Times functionality for Level-8

imkwokyong committed:
Update test case

imkwokyong committed:
Merge branch 'branch-Level-8'

imkwokyong committed:
Add abstract class Command.java

imkwokyong committed:
Modify createList method in Storage class

imkwokyong committed:
Modify code to be more OOP

imkwokyong committed:
Merge branch 'branch-A-MoreOOP'

imkwokyong committed:
Organize the classes into packages

imkwokyong committed:
Merge branch 'add-gradle-support' of https://github.com/imkwokyong/ip into branch-A-Gradle

imkwokyong committed:
Merge branch 'branch-A-Gradle'

imkwokyong committed:
Update main class filepath

imkwokyong committed:
Fix grammar

imkwokyong committed:
Handle more exceptions

imkwokyong committed:
Add JUnit tests for Deadline and Event class

imkwokyong committed:
Add JavaDoc comments

imkwokyong committed:
Merge branch 'branch-A-JavaDoc'

imkwokyong committed:
Add Find functionality for Level-9

imkwokyong committed:
Merge 'branch-Level-9' to master after fixing merge conflicts

imkwokyong committed:
Fix spacing inconsistencies

imkwokyong committed:
Add JavaDoc comments to new methods and classes

imkwokyong committed:
Add checkstyle config

imkwokyong committed:
Adhere to checkstyle requirements

imkwokyong committed:
Merge branch 'branch-A-CodingStandard'

imkwokyong committed:
Fix minor typos and issues

imkwokyong committed:
Add JavaFX support in build.gradle

imkwokyong committed:
Add Launcher.java

imkwokyong committed:
Add images to be used as icons

imkwokyong committed:
Add DialogBox and MainWindow FXML files

imkwokyong committed:
Add executeGui method to all Commands

imkwokyong committed:
Add getResponseToGui method in Duke.java

imkwokyong committed:
Add DialogBox.java, Main.java, MainWindow.java

imkwokyong committed:
Modify font style in FXML files

imkwokyong committed:
Merge branch 'branch-Level-10'

imkwokyong committed:
Adhere to checkstyle requirements

imkwokyong committed:
Change FXML files version to resolve warnings

imkwokyong committed:
Remove 'final' from non-constants

tiif commented on src/main/java/duke/Duke.java:
LGTM. It is following the coding standard.

wnchan replied:
LGTM. Good OOP!

tiif commented on src/main/java/duke/command/ByeCommand.java:
LGTM. It is following the coding standard.

tiif commented on src/main/java/duke/command/DeleteCommand.java:
LGTM. It is following the coding standard.

tiif commented on src/main/java/duke/task/Event.java:
LGTM it is following the coding standard.

wnchan commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,47 @@
+package duke.command;
```
LGTM!

wnchan commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,49 @@
+     * @param input User input.
+     * @throws DukeException If any error occurs.
+     */
+    public MarkCommand(String input) throws DukeException {
```
Nice naming for method and unique error message which gives personality to your chatbot.

wnchan commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,58 @@
+package duke.parser;
```
Good OOP!

wnchan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+        this.isDone = isDone;
+    }
+
+    public Task(String description) {
```
Could add comments for each constructor.

wnchan reviewed (commented):
LGTM! Good implementation and use of OOP throughout code.

imkwokyong committed:
Change mainClass to include GUI in jar file

imkwokyong committed:
Enable keyword in Parser.java to be uppercase

imkwokyong committed:
Enable assertions in build.gradle

There might be some hidden assumptions in the program where, when
failed, will jeopardize the code.

Enabling assertions allows us to check our code against those
assumptions and catch any errors in runtime.

Let's enable assertions so that we could indicate some hidden bugs in
our code as soon as possible.

imkwokyong committed:
Add assertions in Storage.java and MainWindow.java

For Storage.java, the external file might be edited or corrupted.
For MainWindow.java, we assume that the response that we get from Duke
is not an empty message / null.

Adding assertions here will prevent those anomalies from happening, by
detecting them in runtime and throw AssertionError.

Let's,
* add assertion to Storage.java to ensure status is either "0" or "1".
* add assertion to Storage.java to ensure task is of "T", "D" or "E".
* add assertion to MainWindow.java to ensure response is not null.

imkwokyong committed:
Merge pull request #1 from imkwokyong/branch-A-Assertions

Enable and add assertions

imkwokyong committed:
Improve code quality

imkwokyong committed:
Merge pull request #2 from imkwokyong/branch-A-CodeQuality

Improve code quality

imkwokyong committed:
Add overridden equals method for all Task implementation

imkwokyong committed:
Modify addTask method to check for duplicates

imkwokyong committed:
Merge branch 'branch-C-DetectDuplicates'

imkwokyong committed:
Adhere to checkstyle requirements

imkwokyong committed:
Add some personality to the chatbot

imkwokyong committed:
Improve the code to handle more errors

imkwokyong committed:
Add product screenshot

imkwokyong committed:
Add a User Guide

imkwokyong committed:
Update README.md

imkwokyong committed:
Add line breaks to in between each command in User Guide

imkwokyong committed:
Fix minor bug

imkwokyong committed:
Update User Guide
